cmd/lakectl/cmd/config/config.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/spf13/viper"
+)
+
+type configuration struct {
+	Credentials struct {
+		AccessKeyID     string `mapstructure:"access_key_id"`
+		SecretAccessKey string `mapstructure:"secret_access_key"`
+	}
+	Server struct {
+		EndpointURL string `mapstructure:"endpoint_url"`
+	}
+	Metastore struct {
+		Type string
+		Hive struct {
+			URI string
+		}
+		Glue struct {
+			// TODO(ariels): Refactor credentials to share with server side.
+			Profile         string
+			CredentialsFile string `mapstructure:"credentials_file"`
+			Credentials     *struct {
+				AccessKeyID     string `mapstructure:"access_key_id"`
+				AccessSecretKey string `mapstructure:"access_secret_key"`
+				SessionToken    string `mapstructure:"session_token"`
+			}
+
+			Region    string
+			CatalogID string `mapstructure:"catalog_id"`
+		}
+	}
+}
+
+// Config is the currently-loaded configuration.  Its error state supports being able to run
+// `lakectl config' without a valid configuration.
+type Config struct {
+	configuration
+	err error
+}
+
+// ReadConfig loads according to the current viper configuration into a Config, which will
+// have non-nil Err() if loading fails.
+func ReadConfig() (c *Config) {
+	c = &Config{}
+	c.err = viper.ReadInConfig()
+	if c.err != nil {
+		return
+	}
+	c.err = viper.UnmarshalExact(&c.configuration)
+	return
```

Comment:
no retuned named variables, also do not need this one

Reply:
Yeah, you'd think so, but [computer says no](https://play.golang.org/p/uZd7uOFqd4X).  Also [the spec says](missing return at end of function):

>  If the function's signature declares result parameters, the function body's statement list must end in a terminating statement.

Of course it does.